praxis-isncsci/ui
# Dynamically change the key points diagram colors based on the exam values

Closes #185 

## Problem

Add color to the key points diagram to reflect the values being captured.

<img alt="key sensory diagram colors" width="300" src="https://github.com/praxis-isncsci/ui/assets/1294355/94689f06-eb4c-4a77-bcb0-d65894acf030" />

## Solution

- Added a controller which listens to the state changes and modifies the diagram by directly updating the `fill` property of the diagram's segments
- When changing the colors, the controller uses css variables so they can be updated by the design system
- The diagram is now set using an HTML `slot` to give developers more flexibility on what diagram to use and to make it easier to be selected at the document's level

## Testing

- [ ] Can set the color of a segment

<details>
  <summary>Test case</summary>

  1. Navigate to the [key points diagram **Storybook** story](https://64f8d7c6e093108e99084a70-bidiuthobz.chromatic.com/?path=/story/webcomponents-key-points-diagram--primary)
  2. Using the drop-downs at the bottom, select a side, and a color
  3. Press the `set color` button
  4. Confirm that the segment changes to the specified color

</details>

---

- [ ] Can change the segment colors when entering exam values

<details>
  <summary>Test case</summary>

  1. Navigate to the [app **Storybook** story(https://64f8d7c6e093108e99084a70-bidiuthobz.chromatic.com/?path=/story/app-app--primary)
  2. Enter exam values
  3. Confirm that the colors on the diagram update as values are entered for the different key points

</details>
